lib/hanami/providers/db.rb

```diff
@@ -19,6 +19,8 @@ class DB < Dry::System::Provider::Source
 
       # @api private
       def prepare
+        prepare_and_import_parent_db and return if import_from_parent?
```

Comment:
With a few steps to go, it's clear the code this DB provider is going to get somewhat complex.

Since this PR effectively introduces the "second mode" of DB setup (sharing with the app, instead of slice independence), I felt this guard clause + helper method + early return was the most reasonable way to do it.

This keeps the "import"-related functionality together lower in the file, and allows the rest of the DB setup (which is important, because it still has to run at least once) to remain inside the main body of each of the provider lifecycle methods.